cudax/include/cuda/experimental/__memory_resource/async_memory_pool.cuh

```diff
@@ -24,10 +24,10 @@
 // cudaMallocAsync was introduced in CTK 11.2
 #if !defined(_CCCL_COMPILER_MSVC_2017) && !defined(_CCCL_CUDACC_BELOW_11_2)
 
-#  if !defined(_CCCL_CUDA_COMPILER_NVCC) && !defined(_CCCL_CUDA_COMPILER_NVHPC)
```

Comment:
Q: Isn't this an inversion of the previous check? Before we checked whether we are not ony of two CUDA compilers. No we check whether we are a CUDA compiler.

May I assume the previous check was wrong, and this is a fix rather than a refactoring?

Reply:
It generally seems those kinds of changes are independent of the addition `minimum` and `maximum`. Please consider a separate PR in the future, if this was a cleanup.

Reply:
Yeah I was considering that as well, but the PR also changed functional, which broke thrust.

I believe the previous condition was a microoptimization to only include that header for clang-cuda, because that is the only other cuda compiler. The other compilers should have auto-included that already.

The issue is that how it is written it would also be included for any non-cuda compiler.

We can change it to only be included in clang-cuda though

Reply:
Pulled out into a separate PR #2704

Reply:
Ah, I understand. Would have been easier to read if it said something like `#ifdef _CCCL_CLANG_IN_CUDA_MODE`. In any case, I understand the new intent now of only pulling the headers in on a CUDA compiler. LGTM!